config/store.go

```diff
@@ -184,6 +184,10 @@ func (s *Store) Set(newCfg *model.Config) (*model.Config, *model.Config, error)
 	// data from the existing config as necessary.
 	desanitize(oldCfg, newCfg)
 
+	// We apply back environment overrides since the input config may or
+	// may not have them applied.
+	newCfg = applyEnvironmentMap(newCfg, GetEnvironment())
```

Comment:
I'd probably consider moving the call to `fixConfig` as well to make this change more backwards compatible.

Reply:
Hmm, this might change what gets saved to the DB. Are you sure it should be fine? Although this leads to more correct config being stored in the DB, but just concerned about breaking any other thing.

Reply:
That's a good point. The only problem I see here is that what wasn't failing before may start failing here due to this change. Specifically the edge case in which the environment overrides have something to be fixed. Calling `fixConfig` seems like a safe thing to do even though it could lead to something slightly different than the input getting saved.